### DISCUSSION TITLE
Release 1.7.0

### DISCUSSION THREAD

**Discussion by whalley** [moneymanagerex/moneymanagerex#6369] [Development]
@vomikan @n-stein 
Time to tidy-up, hold off on new non-essential commits, and prepare for the next release...
https://github.com/moneymanagerex/moneymanagerex/milestone/45

**Comment by n-stein** [comment]
Roger. Just opened one final PR to finish off #5439.

**Comment by whalley** [comment]
Have set latest release to RC.1 in anticipation of next release before end of year.

**Reply by vomikan** [reply]
The translations look relatively good.
https://crowdin.com/project/moneymanagerex

**Comment by whalley** [comment]
v1.7.0 is released.

@joshua-stone  Could you update Flathub please.

**Reply by joshua-stone** [reply]
[I tried updating the Flathub build](https://github.com/flathub/org.moneymanagerex.MMEX/pull/15), but it would appear that there's an additional layer of validation in Flathub which caused the updated build to fail:

```
{
    "errors": [
        "appstream-failed-validation"
    ],
    "warnings": [
        "appstream-missing-developer-name"
    ],
    "appstream": [
        "style-invalid         : <release> description should be prose and not contain hyperlinks [<p>This release fixes the following bugs:</p><ul><li>#2608 MMEX does not show graphics anymore</li><li>#2601 Top Withdrawals in Home Page</li><li>#2579 Added years for Cash Flow report labels</li><li>#2571 Navigation tree settings ignore some collapsed items</li><li>#2561 Saving of report settings</li><li>#2557 Detected memory leaks on Custom dates</li><li>#2556 #2549 #2554 Stocks Performance report</li><li>#2554 Stock name</li><li>#2520 Fuzzy translations</li><li>#2509 0,00 marked as negative</li><li>#2501 Reports panel</li><li>#2500 Report a bug does not run in an external browser</li><li>#2499 Attachments folder exception</li><li>#2497 Edit Share transaction</li><li>#2496 Unused category</li><li>#2494 Future transactions</li><li>#2492 Account naming issue</li><li>#2488 Attachments folder fix</li><li>#2487 Datepicker appearance</li><li>#2485 Crash on create account for shares</li><li>#2484 Crash or assert on transaction link</li><li>#2480 Display the Budget Setup without category summaries</li><li>#2478 Categories selection</li><li>#2477 wxBitmap labels and sizing issue</li><li>#2470 Crash on every reports Ubuntu 20.04</li><li>#2456 Currency rates fixes</li><li>#2449 Categories do not import when importing QIF files</li><li>#2443 Web app lost category and sub-category</li><li>#2440 #2449 #2443 Web app</li><li>#2433 Expenses report issue</li><li>#2432 Category Income/Expenses - Last 12 Months</li><li>#2427 Ignore future</li><li>#2426 Default Payee in Options/Others</li><li>#2417 Transaction Filter does not open</li><li>#2416 Crash when entering next occurrence of a recurring transaction</li><li>#2413 Webapp autostart memory leaks</li><li>#2410 Notes fiels in CSV import</li><li>#2403 Account in Transaction report selected</li></ul><p>This release adds the following features:</p><ul><li>#2518 Swedish translation</li><li>#2512 Use currency history</li><li>#2506 Account webpage</li><li>#2506 Account webpage</li><li>#2425 Icon sizes</li><li>#2425 Icon sizes</li><li>#2392 Transaction colour</li></ul><p>Currency Exchange Rates</p><p>Virtual File System</p><p>Foreign language versions updated [crowdin.com](https://crowdin.com/project/moneymanagerex/activity_stream)</p>]",
        "style-invalid         : <release> description should be prose and not contain hyperlinks [<p>Your database will be updated once you run this version. You will then not be able to open your database with earlier versions of the software or run Android versions with the same database. As ever, it is recommended to take a copy of your database prior to installing any new software version.</p><p>#1477 changes the DB schema for the storage of categories. Whilst this will not affected basic usage of MMEX if you have created GRM reports that use categories then you may need to update them. The standard GRM templates have been updated to accommodate the new schema and can be used as examples. In particular see: https://github.com/moneymanagerex/general-reports/tree/master/packages/Category/OneCategoryList</p><p>This release adds the following features:</p><ul><li>#1477 Nested Categories</li><li>#5302 Upgrade SQLite</li></ul><p>This release fixes the following bugs:</p><ul><li>#5334 QIF Import does not import transactions with date &lt; today if the account doesn't already exist</li><li>#5346 Account summary not showing assets or shares</li><li>#5352 Crash on selecting Monthly or Yearly - Reports: Summary of Accounts</li><li>#5356 README links to Travis CI for linux/mac build status</li><li>#5359 Order by 'status icon' column in transaction lists</li><li>#5360 Hovering mouse pointer over an asset icon displays unwanted tooltip outline</li><li>#5364 Multi-transaction edit not allowing change to Amount, Payee, or Category</li><li>#5370 Reconciled amount incorrect if deleted transactions exist</li><li>#5372 Resolve MSVC compiler warnings</li><li>#5377 Modifications of split values are lost when row is added</li><li>#5381 Asset - Initial Value and Current Value</li><li>#5382 Mouse wheel scrolling rate is very slow in Split Transactions dialog</li><li>#5391 Use of |</li><li>#5392 Unexpected space added to Category filter when Regex is used to include multiple sub categories</li><li>#5415 MMEX Options: center labels under icons</li><li>#5416 Currency Dialog - Incorrect Column Header</li><li>#5433 Failure to upgrade database to DB v17 for MMEX 1.6.2</li></ul>]",
        "style-invalid         : <release> description should be prose and not contain hyperlinks [<p>This release fixes the following bugs:</p><ul><li>Programs and Features icon in Windows. SF Issue https://sourceforge.net/p/moneymanagerex/bugs/972/</li><li>Albanian lang file</li><li>Prevent Crash on Home page Issue #1019</li><li>Hebrew, Hungarian Translations</li><li>Displayed Latin Currency symbols Issue #739</li><li>Crash on startup when files not found Issue #1033</li><li>Transaction Filter Crash, Account checked, none supplied. Issue #1043</li><li>Check for null pointer. Issue #979</li><li>Move Transaction. Issue #1063</li><li>HTML report export. Issue #1064</li></ul>]"
    ]
}
```

@whalley Can you please remove hyperlinks from the NEWS file and then tag a v1.7.1 hotfix release?

**Reply by joshua-stone** [reply]
We may have to add a call to `appstreamcli validate --strict org.moneymanagerex.MMEX.metainfo.xml` to CMakeLists.txt to ensure that the generated file can pass Flathub validation.

**Reply by whalley** [reply]
> @whalley Can you please remove hyperlinks from the NEWS file and then tag a v1.7.1 hotfix release?

What hyperlinks?

**Reply by whalley** [reply]
Ignore found it. I was looking athe the v1.7.0 detail, but it is in the earlier releases.

**Reply by whalley** [reply]
@joshua-stone  Can you please try this... https://app.circleci.com/pipelines/github/moneymanagerex/moneymanagerex/1166/workflows/88e9aa55-6d94-4581-8028-693331e493a9/jobs/1160. If that works OK I'll tag the commit as 'v1.7.0-Linux-Flatpak-fix'

**Reply by joshua-stone** [reply]
@whalley Thank you! I've updated the build to point to the latest commit, so no tagging required:

https://github.com/flathub/org.moneymanagerex.MMEX/commit/02a34be5985f63b14ba8ee64bcef42a11c38ce78

**Reply by whalley** [reply]
Great. Have tagged anyway.

**Comment by TulioLazarini** [comment]
Hi guys, I'd like to give a very positive feedback about the new version. I use MMEX on a daily basis to manage my personal finances, and the tool is getting better over time because of all your hard work. Well done you all! Thanks to all dev team and wishing you all a Happy New Year. Cheers!

**Comment by Matis1702** [comment]
Buongiorno, vorrei sapere se sarà prevista anche un app Android compatibile con il nuovo database.
Grazie e buon anno

**Reply by whalley** [reply]
Yes - https://github.com/moneymanagerex/android-money-manager-ex/releases/tag/2023.12.27.1007

**Comment by svendickman** [comment]
Hello,
First of all, I love your program and have been using it for years. Since the arrival of the Android https://github.com/moneymanagerex/android-money-manager-ex/releases version compatible with v1.7 I decided to upgrade.
The upgrade works fine on my android, but I cannot open the database on my Windows PCs (I've tried 2 different Win10) and have the exact same error.
I've tried the 32bit and 64bit portable versions. The upgrade of the DB seems to perform correctly, but when I try to open the upgraded DB it stops with in the XML debug file.
The DB can be opened with Android version. If I export the DB in the android I still have the same results.

I have opened an [issue ] https://github.com/moneymanagerex/moneymanagerex/issues/6449
[MoneyManagerEx_dbgrpt-27208-20240110T110604.zip](https://github.com/moneymanagerex/moneymanagerex/files/13887956/MoneyManagerEx_dbgrpt-27208-20240110T110604.zip)


Thanks for any help you can provide
Sven

**Reply by whalley** [reply]
Not sure why you have raised in here as you already have a issue raised here: https://github.com/moneymanagerex/moneymanagerex/issues/6449

**Reply by svendickman** [reply]
After some long trials and errors in deleting certain transactions and accounts I have narrowed the error down to something in stocks and shares.  After removing some portfolios and then upgrading the DB the app no longer crashes on startup.  
  If developers are interested in trying to debug the issue I can try and isolate the issue and share a simplified DB.

re: whalley:  Not sure why you have raised in here as you already have a issue raised here: https://github.com/moneymanagerex/moneymanagerex/issues/6449
         I wasn't sure if the people looking into issues were the same as the people in the discussion forum.  I mentioned the issue in this discussion.

Thanks for your interest and help
Sven

**Reply by whalley** [reply]
> If developers are interested in trying to debug the issue I can try and isolate the issue and share a simplified DB.

Yes please.

**Comment by svendickman** [comment]
Hello Mark,
  I sent a simplified, cleaned up database to the support email address.
There may be some personal info so please limit it's shareing.

  I thought you might enjoy today's https://xkcd.com/2881/

 Thanks again for your implication.
  Feel free if you have any questions.
Sven

**Reply by whalley** [reply]
> I thought you might enjoy today's https://xkcd.com/2881/

I like it!

**Comment by svendickman** [comment]
Good, let me know about the dates for the beach house.

Thanks for your work on what I think is a pretty great application.

Are you also involved with the Android version?

Sven

**Reply by whalley** [reply]
No, not involved in the Android version.

**Comment by svendickman** [comment]
ok,
out of curiosity are you this https://www.linkedin.com/in/markrwhaley/?

**Reply by whalley** [reply]
Nope, this one: https://www.linkedin.com/in/whall3y/
